business/layer.go

```diff
@@ -51,7 +51,7 @@ func NewWithBackends(k8s kubernetes.IstioClientInterface, prom prometheus.Client
 	temporaryLayer.Health = HealthService{prom: prom, k8s: k8s}
 	temporaryLayer.Svc = SvcService{prom: prom, k8s: k8s, businessLayer: temporaryLayer}
 	temporaryLayer.IstioConfig = IstioConfigService{k8s: k8s}
-	temporaryLayer.Workload = WorkloadService{k8s: k8s}
+	temporaryLayer.Workload = WorkloadService{k8s: k8s, prom: prom, ns: NewNamespaceService(k8s)}
```

Comment:
I think you might want to use here the temporaryLayer instead to create a new one.
In similar code as 
```
	temporaryLayer.Svc = SvcService{prom: prom, k8s: k8s, businessLayer: temporaryLayer}
```
businessLayer is passed inside SvcService to access Health service, for WorkloadService you need this to access NamespaceService, I think we can use the same pattern instead to create a new ns variable.

Reply:
We also have the line right below following this other pattern. Besides that, adding the whole layer will add recursion into the structure. For example, if you are using `SvcService` you would be able to call all the `SvcService` public methods from `businessLayer`. IMO, we can avoid this behavior.
Moreover, It sounds to me a little bit overkiller to add the whole other services if you just need one more.

Following what you suggest about avoiding create a new instance, probably I should use the instance created in L57.
```go
temporaryLayer.Workload = WorkloadService{k8s: k8s, prom: prom, ns: temporaryLayer.Namespace}
```
what do you think about that?

Reply:
I would say to follow the same pattern as in SvcService.
We have a mixed of styles in these packages and we should normalize.
It will be refactored in the future all at same time for consistency, but let's not create more combinations of possibilities, I think if there is an example in the same file for what you want to do, the best choice is to replicate it.